functions_lab.js

```diff
@@ -0,0 +1,418 @@
+// | 100 | A+ |
+// | 90 - 99 | A |
+// | 80 - 89 | B |
+// | 70 - 79 | C |
+// | 65 - 69 | D |
+// | Below 65 | F |
+//
+//
+// Example:
+//
+// Input: `88`
+//
+// Output: `"B"`
+//
+// console.log('=======answer#5=========')
+let num = 88
+const letterGrade = (num) => {
+  if (num >= 100) {
+    return "A+"
+  } else if (num <=99  && num >= 90){
+    return "A"
+  } else if (num >= 80 && num <= 89){
+    return "B"
+  } else if (num >= 70 && num <= 79) {
+    return "C"
+  } else if (num >= 65 && num <= 69){
+    return "D"
+  } else {
+    return "F"
+}
+}
+// let num = 88
+let result = letterGrade(num)
+
+
+console.log(result)
+
+
+// ## Question 6
+//
+// Implement a function named `repeatLog` that takes a string and a number
+//as parameters. The function should log `message` `count` number of times.
+//
+// Example:
+//
+// Input: `"+", 10`
+//
+// Output: `++++++++++`
+
+console.log('==========answer#6=======')
+// //
+function repeatLog (param1) {
+  return x = x.repeat(param1)
+  }
+let x = "+"
+let result = x.repeat(10)
+
+  console.log (result)
+//
+//
+// ## Question 7
+//
+// Write a function named `first` that takes a number and returns an array with
+// all the numbers from 1 to n
+//
+//
+// Example:
+//
+// Input: `first(3)`
+//
+// Output: `[1, 2, 3]`
+
+console.log('=========answer#7===========')
+//
+// const first = (num) => {
+//   return num[num.length]
+//   }
+//
+// let num = [3]
+// num.unshift(1, 2);
+//
+// console.log(num)
+//
+//
+// ## Question 8
+//
+// Write a function that logs the numbers from 1 to x, except:
+//
+// If the number if a multiple of 3, log `"Fizz"` instead of the number
+// If the number is a multiple of 5, log `"Buzz"` instead of the number
+// If the number is a multiple of 3 AND 5, log `"FizzBuzz"` instead of the number
+// Your function should take in one parameter: x (the number to count up to)
+//
+console.log('=========answer#8=========')
+
+function multiples (num) {
+  let num2 = []
+for (let i = 1; i <= num; i++) {
+  if (i % 3 === 0 && i % 5 === 0) {
+    num2.push("FizzBuzz")
+} else if (i % 3 === 0) {
+  num2.push("Fizz")
+} else if (i % 5 === 0) {
+    num2.push("Buzz")
+} else {
+    num2.push(i)
+}
+
+  }
+  return num2
+}
+
+console.log(multiples(90))
+//
+// ## Question 9
+//
+// Write a function named `reverse` that takes an array as a parameter.
+//The function should return an array in reverse order.
+//
+//
+// Example:
+//
+// Input: `reverse([1, 2, 3])`
+//
+// Output: `[3, 2, 1]`
+
+console.log('=========answer#9===========')
+let arr = [1, 2, 3];
+
+const reverse = (arr) => {
+let reverseArr = [];
+for (let num of arr) {
+reverseArr.unshift(num)
+
+}
+return reverseArr
+}
+console.log(reverse(arr))
+
+
+
+
+//
+//
+// ## Question 10
+//
+// Write a function that prints out the most frequently occurring element in an
+//array.
+//
+// Example:
+//
+// Input: `[2,1,4,2,5,2,3,23,3,3,2,2,2,24,56,2,2,5,46,2]`
+//
+// Output: `2`
+ console.log("========answer#10=========")
```

Comment:
Please re-visit this question and try to finish it. It is difficult but a must-know question since lots of whiteboarding sessions require solving problems like this.